presidio-analyzer/presidio_analyzer/predefined_recognizers/gliner_recognizer.py

```diff
@@ -0,0 +1,167 @@
+import json
+import logging
+from typing import Dict, List, Optional
+
+from presidio_analyzer import (
+    AnalysisExplanation,
+    LocalRecognizer,
+    RecognizerResult,
+)
+from presidio_analyzer.nlp_engine import NerModelConfiguration, NlpArtifacts
+
+try:
+    from gliner import GLiNER, GLiNERConfig
+except ImportError:
+    GLiNER = None
+    GLiNERConfig = None
+
+
+logger = logging.getLogger("presidio-analyzer")
+
+
+class GLiNERRecognizer(LocalRecognizer):
+    """GLiNER model based entity recognizer."""
+
+    def __init__(
+        self,
+        supported_entities: Optional[List[str]] = None,
+        name: str = "GLiNERRecognizer",
+        supported_language: str = "en",
+        version: str = "0.0.1",
+        context: Optional[List[str]] = None,
+        entity_mapping: Optional[Dict[str, str]] = None,
+        model_name: str = "urchade/gliner_multi_pii-v1",
+        flat_ner: bool = True,
+        multi_label: bool = False,
+        threshold: float = 0.30,
+        map_location: str = "cpu",
+    ):
+        """GLiNER model based entity recognizer.
+
+        The model is based on the GLiNER library.
+
+        :param supported_entities: List of supported entities for this recognizer.
+        If None, all entities in Presidio's default configuration will be used.
+        see `NerModelConfiguration`
+        :param name: Name of the recognizer
+        :param supported_language: Language code to use for the recognizer
+        :param version: Version of the recognizer
+        :param context: N/A for this recognizer
+        :param model_name: The name of the GLiNER model to load
+        :param flat_ner: Whether to use flat NER or not (see GLiNER's documentation)
+        :param multi_label: Whether to use multi-label classification or not
+        (see GLiNER's documentation)
+        :param threshold: The threshold for the model's output
+        (see GLiNER's documentation)
+        :param map_location: The device to use for the model
+
+
+        """
+
+        if entity_mapping:
+            if supported_entities:
+                raise ValueError(
+                    "entity_mapping and supported_entities cannot be used together"
+                )
+
+            self.model_to_presidio_entity_mapping = entity_mapping
+        else:
+            if not supported_entities:
+                logger.info(
+                    "No supported entities provided, "
+                    "using default entities from NerModelConfiguration"
```

Comment:
I see quite a lot more entotoes than what we saw so far from NER modles, do you want to extend the list of keys following GliNER at some point?

Reply:
Hey, do you mean that we should update the list of entities we have as default? The way I see it, GLiNER's added value is when you provide a custom set of entities, as it can work in a zero-shot way.
With regards to mapping, there's a mapping dict in the class, with the ambition to standardize the entities with Presidio's (where possible)

Reply:
@SharonHart let's take it offline